### ISSUE TITLE
[Request]: Additional function to "Disable ragdoll"

### ISSUE THREAD

**Issue by A5m0deu5** [Yimura/YimMenu#2872]
### Problem

Disabling ragdoll provides many advantages, one of which I consider the ability not to fly off the transport when you are standing on top of it. But you can still move off it, you can be pushed off it by some object.

### Solution

Add an option to this feature: "Sticking to the surface", which appears in another line when u enable it (Like with invisibility - Local visibility) and will allow the character to stay stable on the surface he climbed, counting the transport.

### Reason

On the useful side: if there are 2 or less places in the transport that are already occupied, you can climb on it without any problems and ride around with the players. 
On the fun side: You can play animations while standing on a transport without being afraid to fall off it.

**Comment by Tayipc38** [comment]
Enabling the already present feature `No Ragdoll` does this already? I don't understand what you are seeking.

**Comment by A5m0deu5** [comment]
> Enabling the already present feature `No Ragdoll` does this already? I don't understand what you are seeking.

Yes, this feature exists. If you stand on a vehicle with this function, you will not fall, but you still can be moved off from there by some obstacle or during a sharp turn. I'm seeking for an option that allows u to stay still there, "stick" on it.

**Comment by Yimura** [comment]
We could maybe look into attaching the player to the vehicle until they try to move after which we detach them.

However implementing that feels like asking for janky behaviour.

**Comment by gir489returns** [comment]
I FOR ONE really would like this feature, as I do this a lot. Drinking sprunk on their car while they try fling me off. Being stuck to the car would just piss them off to no end.

However, how do we do the math for this? We'd have to know the delta position of the car's origin versus its hood/trunk/roof/wherever the hell you are standing. 

If you can write the math out, I can implement it.

**Comment by Tayipc38** [comment]
Sounds like just attaching our ped to their car tbh. Should still be possible.

**Comment by TheGreenBandit** [comment]
Could just get the relative position to the car and rotation and set it if the player isnt pressing moving binds no?

**Comment by GreySideMM** [comment]
The larger question might be: How to detect if we are actually standing on a car, and on which? Maybe some natives are available.
The math shouldn't be a large problem. Could only get ugly if the car turns more then 90° as I think its not possible to rotate a ped, or?

**Comment by TheGreenBandit** [comment]
> The larger question might be: How to detect if we are actually standing on a car, and on which? Maybe some natives are available.
> 
> The math shouldn't be a large problem. Could only get ugly if the car turns more then 90° as I think its not possible to rotate a ped, or?

local_player->m_vehicle, put in a if statement checking itself then use ptr_to_handle to convert to a Vehicle type

**Comment by GreySideMM** [comment]
> local_player->m_vehicle, put in a if statement checking itself then use ptr_to_handle to convert to a Vehicle type

That's only for a vehicle you are passenger of.
Collision detection might not help enough, as you could be standing besides it. Or you collide with two at once.
But: The physics engine already needs to know on which object you are standing, otherwise it couldn't handle this. Would need some heavy RE I think.
Another idea could be: Bounding box information are available for (some?) objects. These information would be rather crude but could be sufficient.

**Comment by TheGreenBandit** [comment]
Oh yeah kinda overlooked that

**Comment by A5m0deu5** [comment]
There is a function that allows the vehicle stick to the ground. Is it possible to do the same for a character?

**Comment by A5m0deu5** [comment]
> There is a function that allows the vehicle stick to the ground. Is it possible to do the same for a character?

I tried this feature on a wall riding map. The car could move from one surface to another and did not fall from there.

**Comment by lonelybud** [comment]
i once tried to attach the player ped to vehicle and what happens next is crash. I will like to see if anyone is able to get over it. Attaching vehicle to ped will not crash but it will make ped in charge for moving around

**Comment by Tayipc38** [comment]
> i once tried to attach the player ped to vehicle and what happens next is crash. I will like to see if anyone is able to get over it. Attaching vehicle to ped will not crash but it will make ped in charge for moving around

I used to attach myself to everything, including vehicles of other players.

**Comment by lonelybud** [comment]
> I used to attach myself to everything, including vehicles of other players.

i don't try the weird stuffs since the last ban 🙂

**Comment by xesdoog** [comment]
> i don't try the weird stuffs since the last ban 🙂

@lonelybud If you don't mind me asking, what was the reason for your ban? I have a few ideas in mind for some Lua scripts but some of them are somewhat questionable. No griefing or anything like that but I might be messing with something I shouldn't.

**Comment by gir489returns** [comment]
> > i don't try the weird stuffs since the last ban 🙂
> 
> @lonelybud If you don't mind me asking, what was the reason for your ban? I have a few ideas in mind for some Lua scripts but some of them are somewhat questionable. No griefing or anything like that but I might be messing with something I shouldn't.

They don't exactly tell us... It just tells us we're banned from GTA V for cheating.

**Comment by xesdoog** [comment]
> They don't exactly tell us... It just tells us we're banned from GTA V for cheating.

Your name in my notifications scared the shit out of me 😂 I thought you were gonna tear my ass apart for asking a dumb question
